test/functional/test_framework/address.py

```
@@ -46,23 +45,23 @@ def base58_to_byte(s):
     for c in s:
-        if c == chars[0]:
+        if c == b58chars[0]:
             pad += 1
         else:
             break
     res = b'\x00' * pad + res
 
-    # Assert if the checksum is invalid
-    assert_equal(hash256(res[:-4])[:4], res[-4:])
+    if hash256(res[:-4])[:4] != res[-4:]:
+        raise ValueError('Invalid Base58Check checksum')
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)